### PR TITLE
Issue #2583 - add source to Constraint annotation

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java [hunk 1]
```diff
@@ -336,7 +336,7 @@ private void validate(FHIRPathElementNode elementNode) {
                 if (isAbsolute(url)) {
                     if (FHIRRegistry.getInstance().hasResource(url, StructureDefinition.class)) {
                         constraints = new ArrayList<>(constraints);
-                        constraints.add(createConstraint("generated-ext-1", Constraint.LEVEL_RULE, Constraint.LOCATION_BASE, "Extension must conform to definition '" + url + "'", "conformsTo('" + url + "')", false, true));
+                        constraints.add(createConstraint("generated-ext-1", Constraint.LEVEL_RULE, Constraint.LOCATION_BASE, "Extension must conform to definition '" + url + "'", "conformsTo('" + url + "')", "", false, true));
```
Comment (prb112) [inline fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java:339]: What does the empty string signify ?

Comment (prb112) [reply fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java:339]: answered my own question

**Review by prb112 — APPROVED** [review]
LGTM

File: fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java (hunk 1 above)
Comment (JohnTimm) [reply fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java:339]: That's actually something I was just looking at. Whether I should use UNKNOWN or make up a URL that means "this was created in the validator itself". Any thoughts?

Comment (JohnTimm) [reply fhir-validation/src/main/java/com/ibm/fhir/validation/FHIRValidator.java]: Added a public constant `SOURCE_VALIDATOR` with a value of `http://ibm.com/fhir/validation/FHIRValidator`.

**Pull request merged**